lib/sgtn/pseudo.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: EPL-2.0
+
+module Sgtn
+  module Pseudo # :nodoc:
+    def initialize
+      @source_locale = SgtnClient::LocaleUtil.get_source_locale
+      @prefix = Sgtn.pseudo_prefix || '@@'
+      @suffix = Sgtn.pseudo_suffix || @prefix
+    end
+
+    def get_string!(key, component, locale)
+      if Sgtn.pseudo_mode
+        # source is always available, so actual_locale is same as @source_locale
+        translation, actual_locale = super(key, component, @source_locale)
```

Comment:
for online mode, should set the query parameter 'pseudo' to be 'true' when request the pseudo translation from service?

Reply:
It's unnecessary, client will retrieve 'latest' locale and add pseudo tags. They are same.

Reply:
If so, is the pseudo tag same with service '#@'?

Reply:
You can set tags to '#@' by configuration item `pseudo_prefix` and `pseudo_suffix`.

Reply:
how to identify if the pseudo translation comes from online or offline? seems the tags are same for both modes.

Reply:
It's configurable. You can set tags by mode.

Reply:
Seems not. Is there an example?

 For this setting as example:
     _vip_server: https://g11n-vip-stg-1.eng.vmware.com:8090
     pseudo_mode: true
     pseudo_prefix: "@@"
     pseudo_suffix: "@@"_
If the locale is source_locale, it will get local source, so the pseudo is appended with '@@'(such as @@local_source@@); if the locale is not source_locale, it will get latest source from remote service, but the pseudo is appended with same tag '@@' (such as @@remote_latest@@).

Reply:
In pseudo mode, client will return pseudo translation for all locales.